src/js/core/player.js

```diff
@@ -727,12 +727,57 @@ function name() {
     const country = playerNames.countries.find(row => row[1] >= cRand)[0];
 
     // First name
-    const fnRand = random.uniform(0, playerNames.first[country][playerNames.first[country].length - 1][1]);
-    const firstName = playerNames.first[country].find(row => row[1] >= fnRand)[0];
+    let fnRand = random.uniform(0, playerNames.first[country][playerNames.first[country].length - 1][1]);
+    let firstName = playerNames.first[country].find(row => row[1] >= fnRand)[0];
+
+    // Middle Name, randomly
```

Comment:
This code is confusing. What is a "simply name"? Why do we need these while loops? Why are fnRand and firstName no longer constants, it wasn't good enough to just pick them once? More comments or clearer code would be better. Probably would also be good to move some of this to another function, so this function doesn't become giant.